<launch>
    <env name="ROSCONSOLE_CONFIG_FILE" value="$(find exploration)/cfg/custom_rosconsole.config"/>
    <!-- sensor-based -->
    <!-- <arg name="branch"/> sub topic -->
    <!-- <arg name="pose"/> sub topic -->
    <arg name="pose_log"/> <!-- sub topic -->
    <arg name="end"/> <!-- sub topic (main) -->
    <arg name="debug"/> <!-- param (main) -->
    <arg name="rotation"/> <!-- param (main) -->

    <!-- frontier-based -->
    <arg name="frontier" default="frontier"/> <!-- sub topic -->
    <!-- <arg name="pose"/> sub topic -->
    <!-- <arg name="end"/> sub topic (main) -->
    <!-- <arg name="debug"/> param (main) -->
    <!-- <arg name="rotation"/> param (main) -->

    <!-- movement -->
    <!-- <arg name="scan"/> sub topic -->
    <!-- <arg name="bumper"/> sub topic -->
    <!-- <arg name="velocity"/> pub topic -->
    <!-- <arg name="pose"/> sub topic -->

    <!-- launch -->
    <!-- <arg name="debug" default="false"/> -->
    <arg name="robot_name"/>

    <node pkg="exploration" type="legacy_hybrid_exploration" name="legacy_hybrid_exploration" output="screen">
        <remap from="pose_log" to="$(arg pose_log)"/>
        <remap from="end" to="$(arg end)"/>
        <remap unless="$(arg debug)" from="velocity" to="/$(arg robot_name)/mobile_base/commands/velocity"/>
        <remap from="frontier" to="$(arg frontier)"/>

        <param name="debug" value="$(arg debug)"/>
        <param name="rotation" value="$(arg rotation)"/>
        <param name="movebase_name" value="/$(arg robot_name)/move_base"/>
        <param name="angle_bias" value="60"/>
    </node>

    <node pkg="exploration_support" type="loop_closure_counter" name="loop_closure_counter">
        <param name="odom_frame_id" value="$(arg robot_name)/odom"/>
        <param name="map_frame_id" value="$(arg robot_name)/map"/>
    </node>
</launch>
